### PR TITLE
moved to graphql and variable dates

### PR DESCRIPTION
**Pull request by shindym**
- Moved to Graphql api
- Changed from get request to post
- Pulls description out
- Changed release date to a datetime object
- searches based on todays date and the previous date

### CODE REVIEW

File: pipeline/extract_epic.py [hunk 1]
```diff
@@ -1,75 +1,113 @@
...
     }
-    res = req.get(config["BASE_URL"], headers=headers, timeout=10)
+    res = req.post(config["BASE_URL"], json={"query": query}, headers=headers,timeout=10)
     game_data = res.json()
     games = game_data['data']['Catalog']['searchStore']['elements']
     return games
 
 
-def get_game_title(game_obj):
+def get_game_title(game_obj:dict) -> str:
```
Comment (annalisev) [inline pipeline/extract_epic.py:48]: Nice use of type hints

File: pipeline/extract_epic.py [hunk 2]
```diff
@@ -80,4 +118,5 @@ def get_all_games_details(games_obj):
 if __name__ == "__main__":
     load_dotenv()
     games = get_games_data(environ)
-    games_details = get_all_games_details(games)
+    if len(games):
```
Comment (shindym) [inline pipeline/extract_epic.py:121]:
```suggestion
    if len(games)>0:
```

File: pipeline/extract_epic.py [hunk 3]
```diff
@@ -1,75 +1,113 @@
...
+def get_price(game_obj:dict) -> float:
     """Returns price from game json object."""
     return game_obj['currentPrice']/100
 
 
-def get_release_date(game_obj):
+def get_release_date(game_obj:dict):
     """Returns release date from game json object."""
-    return game_obj['releaseDate']
+    return datetime.fromisoformat(game_obj['releaseDate'][0:-5])
```
Comment (Eibironke) [inline pipeline/extract_epic.py:83]: could just be [:-5] instead of [0:-5]

File: pipeline/extract_epic.py [hunk 4]
```diff
@@ -80,4 +118,5 @@ def get_all_games_details(games_obj):
 if __name__ == "__main__":
     load_dotenv()
     games = get_games_data(environ)
-    games_details = get_all_games_details(games)
+    if len(games)>0:
```
Comment (shindym) [inline pipeline/extract_epic.py:121]:
```suggestion
    if len(games) > 0:
```

File: pipeline/extract_epic.py (hunk 3 above)
Comment (Eibironke) [inline pipeline/extract_epic.py:83]:
```suggestion
    return datetime.fromisoformat(game_obj['releaseDate'][:-5])
```

Comment (shindym) [reply pipeline/extract_epic.py]: resolved in suggestions

File: pipeline/extract_epic.py [hunk 5]
```diff
@@ -1,75 +1,113 @@
...
 
-def get_release_date(game_obj):
+def get_release_date(game_obj:dict):
     """Returns release date from game json object."""
-    return game_obj['releaseDate']
+    return datetime.fromisoformat(game_obj['releaseDate'][0:-5])
 
 
-def get_platform_ids(game_tags):
+def get_platform_ids(game_tags:list[str]) -> list[int]:
```
Comment (hhm970) [inline pipeline/extract_epic.py:86]: Very minor change, but maybe you could do a `dict` object, like

```
{'Windows': 1, 'MacOS': 2}
```

and then make this a global variable, as this is an arbitrary allocation?

**Pull request merged**